redis/src/main/scala/zio/redis/RedisCommand.scala

```diff
@@ -18,18 +18,21 @@ package zio.redis
 
 import zio._
 import zio.redis.Input.{StringInput, Varargs}
+import zio.schema.codec.Codec
 
 final class RedisCommand[-In, +Out] private (val name: String, val input: Input[In], val output: Output[Out]) {
-  private[redis] def run(in: In): ZIO[Redis, RedisError, Out] =
-    ZIO
-      .serviceWithZIO[Redis] { redis =>
-        val command = Varargs(StringInput).encode(name.split(" "))(redis.codec) ++ input.encode(in)(redis.codec)
+  private[redis] def run(in: In): ZIO[RedisEnv, RedisError, Out] = {
+    val zio =
+      for {
+        redis <- ZIO.service[Redis]
+        out   <- redis.executor.execute(resp(in, redis.codec))
+        res   <- ZIO.attempt(output.unsafeDecode(out)(redis.codec))
+      } yield res
+    zio.refineToOrDie[RedisError]
```

Comment:
Was this necessary?

Reply:
Don't understand what was necessary?
if u ask about put encode logic in separate method - yes. Cause I'm using it in redis cluster ask command. Don't know how to do it better with current architecture

Reply:
What i see here is that you replaced the serviceWithZIO approach with an equivalent code written via for comprehension which is not really related to the PR scope. Please correct me if I missed something.

Reply:
The main point was to place logic ```Varargs(StringInput).encode(name.split(" "))(codec) ++ input.encode(in)(codec)``` to separate method to call it from RedisClusterExecutorLive
The refactoring was just cause i thought it's new better approach if you want i cat return serviceWithZIO

Reply:
Factoring the encoding out is perfectly fine, the rest is not necessary so please revert it.

Reply:
done